src/storage_volume.h

```diff
@@ -74,10 +74,10 @@ class StorageVolume : public NLVObject<virStorageVolPtr, StorageVolumeCleanupHan
   NLV_PRIMITIVE_RETURN_WORKER(Delete, virStorageVolPtr, bool);
 
   // ACCESSOR/MUTATOR WORKERS
-  NLV_PRIMITIVE_RETURN_WORKER(GetKey, virStorageVolPtr, std::string);
-  NLV_PRIMITIVE_RETURN_WORKER(GetName, virStorageVolPtr, std::string);
-  NLV_PRIMITIVE_RETURN_WORKER(GetPath, virStorageVolPtr, std::string);
-  NLV_PRIMITIVE_RETURN_WORKER(ToXml, virStorageVolPtr, std::string);
+  NLV_STRING_RETURN_WORKER(GetKey, virStorageVolPtr, std::string);
```

Comment:
if you take a look at the conversation page he actually addressed this when I brought it up earlier. I think it would be nice to stick with the primitive macros/template classes with some specialization, but am okay atm with taking this in as it is given that there is limited time for work here